krunner-pass build failure

## Issue description

After switching to 19.03, krunner-pass failed to build

### Steps to reproduce

`nix-shell -p krunner-pass` on 19.03, it is failing with:

```
[ 88%] Building CXX object CMakeFiles/krunner_pass.dir/pass.o
[100%] Linking CXX shared module krunner_pass.so
/nix/store/0y7jmqnj48ikjh37n3dl9kqw9hnn68nq-binutils-2.31.1/bin/ld: cannot find -lQt5::DBus
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/krunner_pass.dir/build.make:99: krunner_pass.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:175: CMakeFiles/krunner_pass.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
builder for '/nix/store/s8a7g0ijn2x50yi5vkkk236022iv1aa3-krunner-pass-1.3.0.drv' failed with exit code 2
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.34, NixOS, 19.03.172138.5c52b25283a (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03.172138.5c52b25283a"`
 - channels(myuser): `"home-manager-19.03"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/tools/security/krunner-pass/default.nix
+++ b/pkgs/tools/security/krunner-pass/default.nix
@@ -1,15 +1,10 @@
-{ mkDerivation, stdenv,
-  fetchFromGitHub,
-  cmake, extra-cmake-modules, gnumake,
+{ mkDerivation, lib, fetchFromGitHub, fetchpatch, cmake, extra-cmake-modules
+, kauth, krunner
+, pass, pass-otp ? null }:
 
-  pass, pass-otp ? null, krunner,
-}:
-let
+mkDerivation rec {
   pname = "krunner-pass";
   version = "1.3.0";
-in
-mkDerivation rec {
-  name = "${pname}-${version}";
 
   src = fetchFromGitHub {
     owner = "akermu";
@@ -19,22 +14,26 @@ mkDerivation rec {
   };
 
   buildInputs  = [
-    pass
-    pass-otp
-    krunner
+    kauth krunner
+    pass pass-otp
   ];
 
-  nativeBuildInputs = [cmake extra-cmake-modules gnumake];
+  nativeBuildInputs = [ cmake extra-cmake-modules ];
 
   patches = [
+    (fetchpatch {
+      url = https://github.com/peterhoeg/krunner-pass/commit/be2695f4ae74b0cccec8294defcc92758583d96b.patch;
+      sha256 = "098dqnal57994p51p2srfzg4lgcd6ybp29h037llr9cdv02hdxvl";
+      name = "fix_build.patch";
+    })
     ./pass-path.patch
   ];
 
   CXXFLAGS = [
-    ''-DNIXPKGS_PASS=\"${stdenv.lib.getBin pass}/bin/pass\"''
+    ''-DNIXPKGS_PASS=\"${lib.getBin pass}/bin/pass\"''
   ];
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     description = "Integrates krunner with pass the unix standard password manager (https://www.passwordstore.org/)";
     homepage = https://github.com/akermu/krunner-pass;
     license = licenses.gpl3;

